lib/src/clojure_lsp/feature/java_interop.clj

```diff
@@ -132,33 +132,22 @@
       (catch Exception e
         (logger/error java-logger-tag "Error Downloading JDK source." e)))))
 
-(defn ^:private analyze-jdk-source! [paths db*]
-  (let [result (shared/logging-time
-                 (str java-logger-tag " Analyzing JDK source with clj-kondo took %s")
-                 (lsp.kondo/run-kondo-on-jdk-source! paths))
-        kondo-analysis (select-keys (:analysis result) [:java-class-definitions])
-        analysis (->> kondo-analysis
-                      (lsp.kondo/normalize-analysis true)
-                      (group-by :filename))]
-    (loop [state-db @db*]
-      (when-not (compare-and-set! db* state-db (update state-db :analysis merge analysis))
-        (logger/warn java-logger-tag "Analyzis outdated from java analysis, trying again...")
-        (recur @db*)))
-    analysis))
-
-(defn ^:private cache-jdk-source-analysis! [analysis new-checksums db*]
-  (loop [state-db @db*]
-    (when-not (compare-and-set! db* state-db (-> state-db
-                                                 (update :analysis merge analysis)
-                                                 (update :analysis-checksums merge new-checksums)))
-      (logger/warn java-logger-tag "Analyzis outdated from java analysis, trying again...")
-      (recur @db*)))
```

Comment:
This also needs a careful review. Originally both `analyze-jdk-source!` and `cache-jdk-source-analysis!` merged the analysis into the db, even though they were called in succession. I made it into one function that analyzes (and saves once) and caches.

Also, I replaced a `loop + compare-and-set!` with `swap!`. My reasoning is that `swap!` does a `loop + compare-and-set!` internally anyway. This loses an error log, but I doubt we would have ever seen that error because the time between the `deref` and the `compare-and-set!` is tiny, and so the `compare-and-set!` will practically always succeed.